File: extract_pdf_text.py
import os
from pypdf import PdfReader
base = r"C:\Users\pietj\Downloads\02_Next_starterkit_Pages_Components-main"
for i in [1,2,3]:
    pdf_path = os.path.join(base, "public", "articles", f"article{i}.pdf")
    out_path = os.path.join(base, f"article{i}_extracted.txt")
    reader = PdfReader(pdf_path)
    parts = []
    for p in reader.pages:
        parts.append(p.extract_text() or "")
    text = "\n\n".join(parts)
    with open(out_path, "w", encoding="utf-8") as f:
        f.write(text)
    print(f"wrote {out_path} ({len(text)} chars)")

File: extract_pdf_text.js
const fs = require('fs');
const path = require('path');
const pdf = require('pdf-parse');

(async () => {
  const base = process.cwd();
  for (const i of [1,2,3]) {
    const pdfPath = path.join(base, 'public', 'articles', `article${i}.pdf`);
    const outPath = path.join(base, `article${i}_extracted.txt`);
    const dataBuffer = fs.readFileSync(pdfPath);
    const data = await pdf(dataBuffer);
    fs.writeFileSync(outPath, data.text || '', 'utf8');
    console.log(`wrote ${outPath} (${(data.text || '').length} chars)`);
  }
})();
